NitroxPatcher/Patches/Dynamic/PDAScanner_Scan_Patch.cs

```diff
@@ -53,6 +63,77 @@ public static void Callback()
             }
         }
 
+        public static readonly int PACKET_SENDING_RATE = 500; // in milliseconds
+
+        // Tuple<LatestProgressTime, LatestPacketSendTime, Entry>
+        public static Dictionary<TechType, Tuple<DateTimeOffset, DateTimeOffset, PDAScanner.Entry>> ThrottlingEntries = new Dictionary<TechType, Tuple<DateTimeOffset, DateTimeOffset, PDAScanner.Entry>>();
+        // Store the thread concerning a certain type
+        public static Dictionary<TechType, Thread> ThrottlingThreads = new Dictionary<TechType, Thread>();
+
+        // After 2 seconds without scanning, a packet will be sent with the latest progress
+        public static void Throttle(TechType techType)
+        {
+            do
+            {
+                Thread.Sleep(2000);
```

Comment:
I don't know for sure but in Unity environments coroutines or the Update function of MonoBehaviors are usually used for things like this. @Measurity do you have more knowledge in this field?

Reply:
Coroutines or tasks are the way to go for short workloads. Thread objects are quite heavy because they need to allocate their own stack and introduce more load on the Windows thread scheduler.